lib/parachain/collation_fetching.go

```diff
@@ -0,0 +1,86 @@
+package parachain
+
+import (
+	"fmt"
+
+	"github.com/ChainSafe/gossamer/lib/common"
+	"github.com/ChainSafe/gossamer/pkg/scale"
+)
+
+// CollationFetchingRequest represents a request to retrieve
+// the advertised collation at the specified relay chain block.
+type CollationFetchingRequest struct {
+	// Relay parent we want a collation for
+	RelayParent common.Hash
+
+	// Parachain id of the collation
+	ParaID ParaID
+}
+
+// Encode returns the SCALE encoding of the CollationFetchingRequest
+func (c *CollationFetchingRequest) Encode() ([]byte, error) {
+	return scale.Marshal(*c)
+}
+
+// CollationFetchingResponse represents a response sent by collator
+type CollationFetchingResponse scale.VaryingDataType
+
+// Collation represents a requested collation to be delivered
+type Collation struct {
+	CandidateReceipt CandidateReceipt `scale:"1"`
+	PoV              PoV              `scale:"2"`
+}
+
+// PoV represents a Proof-of-Validity
+type PoV struct {
+	BlockData []byte
+}
+
+// Index returns the index of varying data type
+func (Collation) Index() uint {
+	return 0
+}
+
+// NewCollationFetchingResponse returns a new collation fetching response varying data type
+func NewCollationFetchingResponse() CollationFetchingResponse {
+	vdt := scale.MustNewVaryingDataType(Collation{})
+	return CollationFetchingResponse(vdt)
+}
+
+// Set will set a value using the underlying  varying data type
+func (c *CollationFetchingResponse) Set(val scale.VaryingDataTypeValue) (err error) {
+	vdt := scale.VaryingDataType(*c)
+	err = vdt.Set(val)
+	if err != nil {
+		return
+	}
+	*c = CollationFetchingResponse(vdt)
+	return
+}
+
+// Value returns the value from the underlying varying data type
+func (c *CollationFetchingResponse) Value() (val scale.VaryingDataTypeValue, err error) {
+	vdt := scale.VaryingDataType(*c)
+	return vdt.Value()
+}
+
+// Encode returns the SCALE encoding of the CollationFetchingResponse
+func (c *CollationFetchingResponse) Encode() ([]byte, error) {
```

Comment:
Again, not sure if these wrappers are needed

Reply:
I have implemented the '[network.ResponseMessage](https://github.com/ChainSafe/gossamer/pull/3334/files#diff-5794e04e36068d98b7c6d2280c7e0b41fecca0b90bd6dd4576cd62b951ab587aR88)' interface as 'CollationFetchingResponse' will be passed into [this](https://github.com/ChainSafe/gossamer/pull/3334/files#diff-5794e04e36068d98b7c6d2280c7e0b41fecca0b90bd6dd4576cd62b951ab587aR32) function as 'res'.